runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -896,15 +896,15 @@ ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *strea
       auto recv = stream->read<uintptr_t, std::string>();
       uintptr_t offset = std::get<0>(recv);
       auto &name = std::get<1>(recv);
-
-      if (offset)
+      auto *vmInfo = TR::compInfoPT->getClientData()->getOrCacheVMInfo(stream);
+      if (offset || (aotCacheLoad && !vmInfo->_hasSharedClassCache))
```

Comment:
This `aotCacheLoad` parameter that I use here should only be temporary - when we ignore the local SCC during fresh JITServer AOT cache compilations this check will need to be updated, because there will be no local offset to store.